integrations/terraform/testlib/role_test.go

```diff
@@ -442,6 +452,19 @@ func (s *TerraformSuiteOSS) TestRoleVersionUpgrade() {
 				Config:   s.getFixture("role_upgrade_v7.tf"),
 				PlanOnly: true,
 			},
+			{
+				Config: s.getFixture("role_upgrade_v8.tf"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, "kind", "role"),
+					resource.TestCheckResourceAttr(name, "version", "v8"),
+					resource.TestCheckResourceAttr(name, "spec.allow.logins.0", "onev8"),
+					checkRoleResource(types.V8, defaultV8Wildcard),
+				),
+			},
+			{
+				Config:   s.getFixture("role_upgrade_v8.tf"),
+				PlanOnly: true,
+			},
```

Comment:
This test fails, the CI is passing because the whole test section is skipped.

Role version conversion is not a new issue and we have the same problem for past versions, but now that we know how to fix it, can we do it before adding another role version? 

Every time we add a version, users are updating and are more likely to hit the inconsistency bug in `kube_resources`.

Reply:
I've created a follow up PR to add a plan_modifier for the `kubernetes_resources` field: https://github.com/gravitational/teleport/pull/62320